### ISSUE TITLE
[Bug] AssertionError triggered with multiple configs from a Config Group specified at CLI

### ISSUE THREAD

**Issue by Jasha10** [facebookresearch/hydra#1446]
# 🐛 Bug
## Description

An assertion error is raised if the CLI is used to select multiple options from a Config Group:
```python
$ cat my_app.py
import hydra
from omegaconf import OmegaConf

@hydra.main()
def my_app(cfg):
    print(OmegaConf.to_yaml(cfg, resolve=True))

if __name__ == "__main__":
    my_app()
```
```bash
$ ls
db              my_app.py       outputs
$ ls db/
mysql.yaml      postgresql.yaml
$ python my_app.py +db=['mysql','postgresql']
Traceback (most recent call last):
  File "/Users/lappy486/hydra/hydra/_internal/utils.py", line 210, in run_and_report
    return func()
  File "/Users/lappy486/hydra/hydra/_internal/utils.py", line 362, in <lambda>
    lambda: hydra.run(
  File "/Users/lappy486/hydra/hydra/_internal/hydra.py", line 86, in run
    cfg = self.compose_config(
  File "/Users/lappy486/hydra/hydra/_internal/hydra.py", line 520, in compose_config
    cfg = self.config_loader.load_configuration(
  File "/Users/lappy486/hydra/hydra/_internal/config_loader_impl.py", line 146, in load_configuration
    return self._load_configuration_impl(
  File "/Users/lappy486/hydra/hydra/_internal/config_loader_impl.py", line 172, in _load_configuration_impl
    defaults_list = create_defaults_list(
  File "/Users/lappy486/hydra/hydra/_internal/defaults_list.py", line 709, in create_defaults_list
    defaults, tree = _create_defaults_list(
  File "/Users/lappy486/hydra/hydra/_internal/defaults_list.py", line 679, in _create_defaults_list
    defaults_tree = _create_defaults_tree(
  File "/Users/lappy486/hydra/hydra/_internal/defaults_list.py", line 338, in _create_defaults_tree
    ret = _create_defaults_tree_impl(
  File "/Users/lappy486/hydra/hydra/_internal/defaults_list.py", line 421, in _create_defaults_tree_impl
    return _expand_virtual_root(repo, root, overrides, skip_missing)
  File "/Users/lappy486/hydra/hydra/_internal/defaults_list.py", line 263, in _expand_virtual_root
    subtree = _create_defaults_tree_impl(
  File "/Users/lappy486/hydra/hydra/_internal/defaults_list.py", line 430, in _create_defaults_tree_impl
    update_package_header(repo=repo, node=parent)
  File "/Users/lappy486/hydra/hydra/_internal/defaults_list.py", line 239, in update_package_header
    loaded = repo.load_config(config_path=node.get_config_path())
  File "/Users/lappy486/hydra/hydra/core/default_element.py", line 470, in get_config_path
    return f"{group_path}/{self.get_name()}"
  File "/Users/lappy486/hydra/hydra/core/default_element.py", line 479, in get_name
    assert self.value is None or isinstance(self.value, str)
AssertionError
```

Note that there is no error if we remove the `db` directory:
```bash
$ rm -rf db/
$ ls
my_app.py       outputs
$ python my_app.py +db=['mysql','postgresql']
db:
- mysql
- postgresql
```

## System information
- **Hydra Version** :  1.1.0dev5
- **Python version** : CPython 3.8
- **Virtual environment type and version** : Conda
- **Operating system** : MacOS

## Additional context
The inspiration for this bug report came from the hydra docs on [Selecting multiple configs from a Config Group](https://hydra.cc/docs/next/patterns/select_multiple_configs_from_config_group/)

**Comment by omry** [comment]
> Note that there is no error if we remove the db directory:

Removing the db directory changes from from a statement overriding the config group db to a list of two configs, into a list of two values (which is a completely different beast).

**Comment by omry** [comment]
Thanks for reporting.
I have a fix in #1456. This is hitting an edge case when a primary config is not defined which is a mode almost no one is using (it mostly exists as an educational tool).
